src/server_manager/web_app/shadowbox_server.ts

```diff
@@ -243,33 +220,32 @@ export class ShadowboxServer implements server.Server {
     }
   }
 
-  setPortForNewAccessKeys(newPort: number): Promise<void> {
+  async setPortForNewAccessKeys(newPort: number): Promise<void> {
     console.info(`setPortForNewAccessKeys: ${newPort}`);
-    const requestOptions: RequestInit = {
-      method: 'PUT',
-      headers: new Headers({'Content-Type': 'application/json'}),
-      body: JSON.stringify({port: newPort}),
-    };
-    return this.apiRequest<void>('server/port-for-new-access-keys', requestOptions).then(() => {
-      this.serverConfig.portForNewAccessKeys = newPort;
-    });
+    await this.apiRequest<void>('server/port-for-new-access-keys', 'PUT', {json: {port: newPort}});
+    this.serverConfig.portForNewAccessKeys = newPort;
   }
 
   private async getServerConfig(): Promise<ServerConfigJson> {
     console.info('Retrieving server configuration');
     return await this.apiRequest<ServerConfigJson>('server');
   }
 
-  protected setManagementApiUrl(apiAddress: string): void {
+  protected setManagementApiUrl(apiAddress: string, fingerprint?: string): void {
     this.managementApiAddress = apiAddress;
+    this.fingerprint = fingerprint;
   }
 
   getManagementApiUrl() {
     return this.managementApiAddress;
   }
 
   // Makes a request to the management API.
-  private apiRequest<T>(path: string, options?: RequestInit): Promise<T> {
+  private apiRequest<T extends object | void>(
```

Comment:
We can be more object-oriented for readability and maintainability.

Perhaps this logic can be extracted into some sort of `HttpEndpoint` class that encapsulated the fingerprint stuff and provides a `HttpEndpoint.request<>()` method that does the `apiRequest` logic.

Then you call `ShadowboxServer.setManagementApiEndpoint(endpoint: HttpEndpoint)`. ShadowboxServer doesn't need to know how the API calls are made. It can also be a way to do testing by injecting fake endpoints.

Reply:
Done.  I called the class `PathApiClient`.